### PR TITLE
Avoiding crash for Edgetrace error

### PR DESCRIPTION
**Pull request by debora-pe**
This PR deals with crashes that happen when slit trace fails because the pca is not possible.
Instead of crashing, now it will result as a failed calibration and move on to the following detector (if exist).

I also added some tweaks to slitmask matching if pca is not available.

Non-related, I updated some documentation.

Tested on datasets that crashed before. I'll run the Dev Suite.

### CODE REVIEW

File: doc/flat_fielding.rst
Comment (profxj) [inline doc/flat_fielding.rst:124]: Did this file not make it in the previous PR (MOSFIRE)?

File: pypeit/edgetrace.py
Comment (profxj) [inline pypeit/edgetrace.py]: We have generally tried to avoid over-riding `par` within the code (after instantiating the `spectrograph`).

But i can see why this is done here.  
Another approach to consider would be to have the EdgeTrace object hold this parameter.  It would
be initialized to the value from `self.par` but we could
over-ride it as done here.

A bit klunky, but would maintain the "avoid editing par" mantra.

File: pypeit/edgetrace.py
Comment (profxj) [inline pypeit/edgetrace.py]: Same point as above.

File: pypeit/pypeit.py
Comment (profxj) [inline pypeit/pypeit.py:317]: Nice fix!

File: pypeit/pypeit.py
Comment (profxj) [inline pypeit/pypeit.py:448]: Not successful for all or for any?

**Review by profxj — CHANGES_REQUESTED** [review]
Only 1 significant item to discuss.  I'll raise it at today's meeting.

File: pypeit/edgetrace.py
Comment (debora-pe) [reply pypeit/edgetrace.py]: I implemented the changes we discussed at the PypeIt meeting

File: pypeit/pypeit.py
Comment (debora-pe) [reply pypeit/pypeit.py:448]: It's "not successful for all". If for some reason, some calibration step failed for all the detectors, `self.save_exposure` would crash.

File: doc/flat_fielding.rst
Comment (debora-pe) [reply doc/flat_fielding.rst:124]: Yes, I realized, I only updated the mosfire doc, but lampoffflats can be used with any spectrograph.

File: pypeit/par/pypeitpar.py
Comment (kbwestfall) [inline pypeit/par/pypeitpar.py]: Very nice.  Can you also make some minor edits to the `Slit PCA fails` section of the `doc/slit_tracing.rst` file?  Basically, I would just indicate that auto is an option.

File: pypeit/spectrographs/keck_deimos.py
Comment (kbwestfall) [inline pypeit/spectrographs/keck_deimos.py:412]: Can you add a sentence or two describing where and why this is called currently in the code?

File: pypeit/spectrographs/spectrograph.py
Comment (kbwestfall) [inline pypeit/spectrographs/spectrograph.py:189]: Same request here for a brief description.

**Review by kbwestfall — APPROVED** [review]
Thanks @deborape !  I just had a couple of (hopefully easy) doc requests.  Approving now.

File: pypeit/par/pypeitpar.py
Comment (debora-pe) [reply pypeit/par/pypeitpar.py]: Thanks! Done

File: pypeit/spectrographs/keck_deimos.py
Comment (debora-pe) [reply pypeit/spectrographs/keck_deimos.py:412]: Done

File: pypeit/spectrographs/spectrograph.py
Comment (debora-pe) [reply pypeit/spectrographs/spectrograph.py:189]: Done

**Review by profxj — APPROVED** [review]
Approved!

**Comment by debora-pe** [comment]
Finally I run the tests for this. All good!

```
--- PYPEIT DEVELOPMENT SUITE PASSED 151/151 TESTS (Masters ignored) ---
Testing Started at 2022-06-18T19:53:46.711109
Testing Completed at 2022-06-19T04:02:25.466793
Total Time: 8:08:38.755684
```

```
pytest . -W ignore
============================= test session starts =============================
platform darwin -- Python 3.9.7, pytest-7.1.1, pluggy-1.0.0
rootdir: /Users/dpelliccia/PypeIt/PypeIt, configfile: setup.cfg
collected 329 items                                                           

test_alignments.py .                                                    [  0%]
test_arc.py .                                                           [  0%]
test_archive.py ..                                                      [  1%]
test_arcimage.py ..                                                     [  1%]
test_biasframe.py ....                                                  [  3%]
test_bitmask.py ....                                                    [  4%]
test_bpmimage.py ..                                                     [  4%]
test_bspline.py ...............                                         [  9%]
test_buildimage.py ..                                                   [ 10%]
test_calibrations.py .......                                            [ 12%]
test_chk_calibs.py ..                                                   [ 12%]
test_coadd.py .                                                         [ 13%]
test_collate_1d.py ..........                                           [ 16%]
test_cooked.py .                                                        [ 16%]
test_coremeta.py .                                                      [ 16%]
test_datacontainer.py .......                                           [ 18%]
test_detector.py ....                                                   [ 20%]
test_display.py .                                                       [ 20%]
test_edgetrace.py ..                                                    [ 20%]
test_fitting.py ....                                                    [ 22%]
test_flatfield.py ..                                                    [ 22%]
test_flexure.py ..                                                      [ 23%]
test_flux.py ....                                                       [ 24%]
test_fluxspec.py .....                                                  [ 26%]
test_frametype.py ..                                                    [ 26%]
test_history.py ....                                                    [ 27%]
test_load.py .                                                          [ 28%]
test_load_images.py ................                                    [ 33%]
test_manual.py ..                                                       [ 33%]
test_maskdesign_matching.py ...                                         [ 34%]
test_masterframe.py .                                                   [ 34%]
test_match.py ....                                                      [ 36%]
test_metadata.py ......                                                 [ 37%]
test_moment.py ......                                                   [ 39%]
test_mosaic.py ...                                                      [ 40%]
test_msgs.py ..                                                         [ 41%]
test_opticalmodel.py ....                                               [ 42%]
test_parse.py ...                                                       [ 43%]
test_pca.py ........                                                    [ 45%]
test_pkgdata.py ...                                                     [ 46%]
test_processimage.py ...                                                [ 47%]
test_procimg.py ........                                                [ 50%]
test_pydl.py .                                                          [ 50%]
test_pypeit.py .                                                        [ 50%]
test_pypeitimage.py ..                                                  [ 51%]
test_pypeitpar.py ................................                      [ 61%]
test_pypeitsetup.py ........                                            [ 63%]
test_qa.py .                                                            [ 63%]
test_runpypeit.py ..                                                    [ 64%]
test_scienceimage.py ...                                                [ 65%]
test_scripts.py ....................                                    [ 71%]
test_sensfilearchive.py ...                                             [ 72%]
test_setups.py ......................                                   [ 79%]
test_skysub.py ..                                                       [ 79%]
test_slitmask.py ..                                                     [ 80%]
test_slittrace.py ....                                                  [ 81%]
test_spec2dobj.py ......                                                [ 83%]
test_specobj.py .......                                                 [ 85%]
test_specobjs.py .....                                                  [ 86%]
test_spectrographs.py .......................                           [ 93%]
test_syncspec.py .                                                      [ 94%]
test_telluric.py ..                                                     [ 94%]
test_traceimage.py .                                                    [ 95%]
test_transform.py ...                                                   [ 96%]
test_utils.py ....                                                      [ 97%]
test_vcorr.py ..                                                        [ 97%]
test_wavemodel.py .                                                     [ 98%]
test_wavetilts.py ....                                                  [ 99%]
test_wvcalib.py ..                                                      [100%]

====================== 329 passed in 1295.93s (0:21:35) =======================
```

**Comment by debora-pe** [comment]
merging

**Pull request merged**
